controllers/cloud.redhat.com/providers/kafka/strimzi.go

```diff
@@ -210,46 +227,68 @@ func (s *strimziProvider) configureKafkaConnectCluster() error {
 }
 
 func (s *strimziProvider) configureListeners() error {
-	// clusterNN := types.NamespacedName{
-	// 	Namespace: getKafkaNamespace(s.Env),
-	// 	Name:      s.Env.Spec.Providers.Kafka.Cluster.Name,
-	// }
-	// kafkaResource := strimzi.Kafka{}
-	// if _, err := utils.UpdateOrErr(s.Client.Get(s.Ctx, clusterNN, &kafkaResource)); err != nil {
-	// 	return err
-	// }
-
-	// // Return an err if we can't obtain listener status to trigger a requeue in the env controller
-	// if kafkaResource.Status == nil || kafkaResource.Status.Listeners == nil {
-	// 	return fmt.Errorf(
-	// 		"Kafka cluster '%s' in ns '%s' has no listener status", clusterNN.Name, clusterNN.Namespace,
-	// 	)
-	// }
+	clusterNN := types.NamespacedName{
+		Namespace: getKafkaNamespace(s.Env),
+		Name:      s.Env.Spec.Providers.Kafka.Cluster.Name,
+	}
+	kafkaResource := strimzi.Kafka{}
+	if _, err := utils.UpdateOrErr(s.Client.Get(s.Ctx, clusterNN, &kafkaResource)); err != nil {
+		return err
+	}
+
+	// Return an err if we can't obtain listener status to trigger a requeue in the env controller
+	if kafkaResource.Status == nil || kafkaResource.Status.Listeners == nil {
+		return fmt.Errorf(
+			"Kafka cluster '%s' in ns '%s' has no listener status", clusterNN.Name, clusterNN.Namespace,
+		)
+	}
+
+	kafkaCASecName := types.NamespacedName{
+		Name:      fmt.Sprintf("%s-cluster-ca-cert", s.Env.Spec.Providers.Kafka.Cluster.Name),
+		Namespace: getKafkaNamespace(s.Env),
+	}
+	kafkaCASecret := core.Secret{}
+	if _, err := utils.UpdateOrErr(s.Client.Get(s.Ctx, kafkaCASecName, &kafkaCASecret)); err != nil {
+		return err
+	}
+
+	kafkaCACert := string(kafkaCASecret.Data["ca.crt"])
 
 	s.Config.Brokers = []config.BrokerConfig{}
-	s.Config.Brokers = append(s.Config.Brokers, config.BrokerConfig{
-		Hostname: fmt.Sprintf("%s-kafka-bootstrap.%s.svc", s.Env.Spec.Providers.Kafka.Cluster.Name, getKafkaNamespace(s.Env)),
-		Port:     utils.IntPtr(9092),
-	})
-	// for _, listener := range kafkaResource.Status.Listeners {
-	// 	if listener.Type != nil && (*listener.Type == "plain" || *listener.Type == "tcp") {
-	// 		bc := config.BrokerConfig{
-	// 			Hostname: *listener.Addresses[0].Host,
-	// 		}
-	// 		port := listener.Addresses[0].Port
-	// 		if port != nil {
-	// 			p := int(*port)
-	// 			bc.Port = &p
-	// 		}
-	// 		s.Config.Brokers = append(s.Config.Brokers, bc)
-	// 	}
-	// }
-
-	// if len(s.Config.Brokers) < 1 {
-	// 	return fmt.Errorf(
-	// 		"Kafka cluster '%s' in ns '%s' has no listeners", clusterNN.Name, clusterNN.Namespace,
-	// 	)
-	// }
+	for _, listener := range kafkaResource.Status.Listeners {
+		if listener.Type != nil && *listener.Type == "tls" {
+			authType := config.BrokerConfigAuthtypeSasl
```

Comment:
Can we move this to a func called `buildTlsBrokerConfig` which returns a `config.BrokerConfig` ?

Reply:
we can